wordpress-mobile/WordPress-iOS
# AB testing: implement the first A/B experiment

Related PR: https://github.com/Automattic/Automattic-Tracks-iOS/pull/174

After our successful A/A test (https://github.com/wordpress-mobile/WordPress-iOS/pull/15789) it's time for our first A/B test!

This PR implements the first A/B test within the FAB button (under My Site).

Also, it fixes a bug identified in the previous version (which was requesting the `assignments` endpoint even without any active tests)

| Control | Treatment |
| ------- | ----------- |
| ![Simulator Screen Shot - iPhone 11 - 2021-03-18 at 16 22 06](https://user-images.githubusercontent.com/7040243/111685148-51e3d880-8806-11eb-8b32-71809f3fb138.png) | ![Simulator Screen Shot - iPhone 11 - 2021-03-18 at 16 23 23](https://user-images.githubusercontent.com/7040243/111685180-5e683100-8806-11eb-8b90-d43fcb4b7a57.png) |

### To test

1. Open the app
2. Tap the FAB, you should see the following order: "Blog post", "Site page" and "Story post"
3. Close the app

Now you'll need to use a tool like Charles to intercept the proxy.

1. Open the app again
2. Intercept the assignments request and change `wpios_create_menu_story_first` to `treatment`
3. 2. Tap the FAB
4. You should see the following order: "Story post", "Blog post" and "Site page"

You can see this test in the following video:

https://user-images.githubusercontent.com/7040243/111689322-a5582580-880a-11eb-9cda-429392162420.mov



### Exposure event

The PR is still missing the exposure event, I'll add it before merging it.
